contrib/gomodule/redigo/redigo.go

```diff
@@ -110,11 +124,7 @@ func (tc Conn) newChildSpan(ctx context.Context) ddtrace.Span {
 	return span
 }
 
-// Do wraps redis.Conn.Do. It sends a command to the Redis server and returns the received reply.
-// In the process it emits a span containing key information about the command sent.
-// When passed a context.Context as the final argument, Do will ensure that any span created
-// inherits from this context. The rest of the arguments are passed through to the Redis server unchanged.
-func (tc Conn) Do(commandName string, args ...interface{}) (reply interface{}, err error) {
+func withSpan(do func(commandName string, args ...interface{}) (interface{}, error), p *params, commandName string, args ...interface{}) (reply interface{}, err error) {
```

Comment:
```suggestion
func withSpan(do redis.Conn, p *params, commandName string, args ...interface{}) (reply interface{}, err error) {
```
This should work fine, right?

Reply:
hmm that's a good question, if you take a `redis.Conn` here you can't call `DoWithTimeout` right? The thing here is that the preparation of the span is the same no matter what "kind" of `Do` call we do afterwards, but there's no way (that I could see) to make the code generic apart from letting the caller inject the actual calling logic.

Reply:
Yes, of course. You are right.